### PR TITLE
[flang] Implement SECOND intrinsic

### PR DESCRIPTION
**Pull request by tblah**
The SECOND intrinsic is a gnu extension providing an alias for CPU_TIME: https://gcc.gnu.org/onlinedocs/gfortran/SECOND.html

This cannot be implemented as a straightforward alias because there is both a function and a subroutine form.

### CODE REVIEW

File: flang/docs/Intrinsics.md [hunk 1]
```diff
@@ -1047,3 +1047,16 @@ program rename_proc
...
 end program rename_proc
 ```
+
+### Non-standard Intrinsics: SECOND
+This intrinsic is an alias for `CPU_TIME`: supporting both a subroutine and a
+function form.
+
+#### Usage and Info
+
+- **Stnadard:** GNU extension
```
Comment (mjklemm) [inline flang/docs/Intrinsics.md:1057]:
```suggestion
- **Standard:** GNU extension
```

**Review by mjklemm — APPROVED** [review]
LGTM (with nit about typo :-))

**Review by jeanPerier — APPROVED** [review]
Thanks

**Review by clementval — APPROVED** [review]

File: flang/lib/Optimizer/Builder/IntrinsicCall.cpp [hunk 2]
```diff
@@ -6140,6 +6144,27 @@ IntrinsicLibrary::genScan(mlir::Type resultType,
   return readAndAddCleanUp(resultMutableBox, resultType, "SCAN");
 }
 
+// SECOND
+fir::ExtendedValue
+IntrinsicLibrary::genSecond(std::optional<mlir::Type> resultType,
+                            mlir::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 1 && !resultType || args.empty() && resultType);
```
Comment (Thirumalai-Shaktivel) [inline flang/lib/Optimizer/Builder/IntrinsicCall.cpp:6151]: This causes the build to fail, can you please check?
```
[...]
/home/tshaktiv/Projects/llvm-project/flang/lib/Optimizer/Builder/IntrinsicCall.cpp:6151:27: error: suggest parentheses around ‘&&’ within ‘||’ [-Werror=parentheses]
 6151 |   assert(args.size() == 1 && !resultType || args.empty() && resultType);
      |          ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

Comment (tblah) [reply flang/lib/Optimizer/Builder/IntrinsicCall.cpp:6151]: Thanks. See https://github.com/llvm/llvm-project/pull/99306

**Pull request merged**